### PR TITLE
DC-401: Get catalog UI ready for production release

### PR DESCRIPTION
**Pull request by tcjiang9**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "jszip": "^3.7.1",
     "lodash": "^4.17.21",
     "marked": "^4.0.10",
-    "oidc-client-ts": "^2.0.4",
+    "oidc-client-ts": "^2.0.5",
```
Comment (nawatts) [inline package.json]: This looks unrelated to the ticket and other changes. What was the motivation for upgrading this library?

File: src/pages/library/Datasets.js [hunk 2]
```diff
@@ -427,11 +429,24 @@ const topMed = () => h(Participant, {
   }, ['Browse Data'])
 ])
 
+const DatasetsAuthWrapper = () => {
```
Comment (nawatts) [inline src/pages/library/Datasets.js]: What is the purpose of this wrapper?

File: src/pages/library/RequestDatasetAccessModal.js [hunk 3]
```diff
@@ -32,7 +32,7 @@ export const RequestDatasetAccessModal = ({ onDismiss, datasets }) => {
       strong(['Pending']), `.`
     ]),
     div({ style: { marginTop: 30, display: 'flex', flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center' } }, [
-      `Let's continue testing the BETA Data Catalog!`,
+      `Let's continue testing the New Data Catalog!`,
```
Comment (nawatts) [inline src/pages/library/RequestDatasetAccessModal.js]: Nit: should "New" be lowercase?

**Review by nawatts — CHANGES_REQUESTED** [review]
Changes related to the ticket look mostly fine, but there are a lot of ESLint warnings that need to be addressed (run `yarn run lint` to see them). Upgrading the `oidc-client-ts` library seems unrelated to this ticket, so I think it should be removed from this PR.

File: src/pages/library/Datasets.js (hunk 2 above)
Comment (tcjiang9) [reply src/pages/library/Datasets.js]: it's a bit of a hack to ensure the Datasets component rerenders when a new user login completes

File: package.json (hunk 1 above)
Comment (tcjiang9) [reply package.json]: this was done automatically when I ran yarn install, I can undo it

File: src/pages/library/Datasets.js (hunk 2 above)
Comment (nawatts) [reply src/pages/library/Datasets.js]: Ah, because `getLocalPref('catalog-toggle')` depends on the current user ID and needs to be rerun when the user changes?

I think this would be worth a comment since it's not obvious.

Also, I suspect `useStore(authStore)` could be placed in the `Datasets` component instead of a wrapper component. And the return value doesn't need to be stored in a variable.

Comment (nawatts) [reply src/pages/library/Datasets.js]: Then again, since Datasets is a child of Router, which is a child of AuthContainer and AuthContainer subscribes to changes in the auth store, I would expect this to re-render when the user changes even without subscribing to the auth store itself.

File: src/pages/library/Datasets.js [hunk 4]
```diff
@@ -445,8 +461,8 @@ export const navPaths = [
   {
     name: 'library-datasets',
     path: '/library/datasets',
-    component: Datasets,
-    public: true,
+    component: DatasetsAuthWrapper,
+    public: false,
```
Comment (nawatts) [inline src/pages/library/Datasets.js:476]: If this page is going to require login, should it still appear in the sidebar when the user is not logged in?

Screen Shot 2022-06-29 at 2 48 48 PM

File: src/pages/library/Datasets.js (hunk 2 above)
Comment (s-rubenstein) [reply src/pages/library/Datasets.js]: Oh! When we were pairing I missed that the Router will trigger a re-render when the user changes. That's great in this case!

We can't put `useStore(authStore)` in `Datasets`. I think we tried that first and it didn't work, which is why we moved to this approach. We wanted a complete re-instantiation of the component, rather than a partial re-rendering.

File: integration-tests/tests/preview-dataset.js [hunk 5]
```diff
@@ -9,7 +9,7 @@ const datasetName = 'Readable Catalog Snapshot 1'
 const testPreviewDatasetFn = withUserToken(async ({ testUrl, page, token }) => {
   await enableDataCatalog(page, testUrl, token)
   await click(page, clickable({ textContains: 'datasets' }))
-  await click(page, clickable({ textContains: 'BETA Data Catalog OFF' }))
+  await click(page, clickable({ textContains: 'New Data Catalog OFF' }))
```
Comment (s-rubenstein) [inline integration-tests/tests/preview-dataset.js]: Is there a ticket in the backlog to remove the word `New` in all these places? I feel like we're gunna wanna make that change at some point.

File: integration-tests/utils/integration-helpers.js [hunk 6]
```diff
@@ -185,7 +185,6 @@ const enableDataCatalog = async (page, testUrl, token) => {
   await waitForNoSpinners(page)
 
   await findText(page, 'Browse Data')
-  await overrideConfig(page, { isDataBrowserVisible: true })
```
Comment (s-rubenstein) [inline integration-tests/utils/integration-helpers.js:188]: I think these tests might still need a change to flip the switch to turn on the data catalog?

File: src/pages/library/DataBrowserToggler.js [hunk 7]
```diff
@@ -97,22 +92,15 @@ export const DataBrowserPreviewToggler = ({ checked }) => {
...
           width: 55, height: 25,
-          onChange: () => {
-            catalogPreviewStore.set({ [id]: !checked })
-            if (checked) {
-              Nav.goToPath('library-datasets')
-            } else {
-              Nav.goToPath('library-browser')
-            }
-          }
+          onChange: value => onChange(value)
```
Comment (s-rubenstein) [inline src/pages/library/DataBrowserToggler.js]: I know we changed to this here, but I think we might just be able to have this be `onChange`, and it should pick up value?

I tested in the console with the following set of commands:
```
const foo = (val) => console.log(val)
undefined
foo
(val) => console.log(val)
const obj = { foo }
undefined
obj.foo("a")
a
```

File: src/pages/library/Datasets.js [hunk 8]
```diff
@@ -427,16 +429,30 @@ const topMed = () => h(Participant, {
...
 
+const DatasetsAuthWrapper = () => {
+  // unused variable to re-render Datasets
+  const authState = useStore(authStore)
+  return h(Datasets)
+}
 
 const Datasets = () => {
+  const [catalogShowing, setCatalogShowing] = useState(!!getLocalPref('catalog-toggle'))
+  console.log(`DATASETS CAT SHOWING ${catalogShowing}`)
```
Comment (s-rubenstein) [inline src/pages/library/Datasets.js]: We can remove the console log at this point.

File: package.json (hunk 1 above)
Comment (tcjiang9) [reply package.json]: @nawatts 
looks like if I undo this, I get a circleci build failure with
➤ YN0056: │ oidc-client-ts@npm:2.0.5: Cache entry required but missing for oidc-client-ts@npm:2.0.5

File: src/pages/library/Datasets.js (hunk 2 above)
Comment (tcjiang9) [reply src/pages/library/Datasets.js]: @nawatts looks like you're right here, it turned out to not be necessary, thanks!

File: src/pages/library/Datasets.js (hunk 4 above)
Comment (tcjiang9) [reply src/pages/library/Datasets.js:476]: @nawatts, following this logic, should I remove the card for "Browse Data" from the front page also?

Comment (tcjiang9) [reply src/pages/library/Datasets.js:476]: the 'View Workspaces' card also requires login

File: src/pages/LandingPage.js [hunk 9]
```diff
@@ -103,7 +101,7 @@ const LandingPage = () => {
       }, ['See this article']),
       ' for a summary of available resources.'
     ]),
-    isDataBrowserVisible() && div({
+    div({
```
Comment (pshapiro4broad) [inline src/pages/LandingPage.js]: I don't know if we want to show this `div()` yet. It's a change to the terra landing page, and it's a banner that will direct users to the catalog view. I think we want to wait on this until we're ready to more publicly promote the new catalog view. If so, we'll need to keep a feature flag in place to hide this banner.

File: src/pages/library/DataBrowserToggler.js [hunk 10]
```diff
@@ -97,22 +90,15 @@ export const DataBrowserPreviewToggler = ({ checked }) => {
...
         }),
-        div({ style: { marginLeft: 10 } }, [`BETA Data Catalog ${checked ? 'ON' : 'OFF'}`])
+        div({ style: { marginLeft: 10 } }, [`New Data Catalog ${catalogShowing ? 'ON' : 'OFF'}`])
```
Comment (pshapiro4broad) [inline src/pages/library/DataBrowserToggler.js]: It looks like doesn't reflect the latest invision mocks. In the one I see, this text says `New Catalog OFF` or `New Catalog ON`. https://projects.invisionapp.com/d/main?origin=v7#/console/22047689/467561712/preview

File: src/pages/library/RequestDatasetAccessModal.js [hunk 11]
```diff
@@ -21,7 +21,7 @@ export const RequestDatasetAccessModal = ({ onDismiss, datasets }) => {
   const [showWipModal, setShowWipModal] = useState(false)
 
   return showWipModal ? h(Modal, {
-    title: `Keep in mind we're still in BETA`,
+    title: `Keep in mind we're still in development`,
```
Comment (pshapiro4broad) [inline src/pages/library/RequestDatasetAccessModal.js:24]: This is fine although we don't have any immediate plans to add controlled access data to the catalog, and this dialog won't be seen until we do. At that point we'll probably revisit this UI as well.

File: integration-tests/utils/integration-helpers.js (hunk 6 above)
Comment (tcjiang9) [reply integration-tests/utils/integration-helpers.js:188]: looks like  the test to flip the catalog is already performed in other tests
```
const testPreviewDatasetFn = withUserToken(async ({ testUrl, page, token }) => {
  await enableDataCatalog(page, testUrl, token)
  await click(page, clickable({ textContains: 'datasets' }))
  await click(page, clickable({ textContains: 'New Data Catalog OFF' }))
  await waitForNoSpinners(page)

```

I think this test was just to test the config override and so can probably be deleted

File: src/pages/library/DataBrowserToggler.js [hunk 12]
```diff
@@ -97,22 +90,15 @@ export const DataBrowserPreviewToggler = ({ checked }) => {
...
       ])
     ]),
-    checked && div({ style: { display: 'flex' } }, [
+    catalogShowing && div({ style: { display: 'flex' } }, [
       icon('talk-bubble', { size: 45, style: { marginLeft: '1.5rem', margin: '0 0.5rem' } }),
       div({ style: { display: 'flex', flexDirection: 'column' } }, [
         strong('Provide feedback'),
```
Comment (pshapiro4broad) [inline src/pages/library/DataBrowserToggler.js]: The feedback text here has also been updated in the latest UI mocks https://projects.invisionapp.com/d/main?origin=v7#/console/22047689/467561713/preview

**Review by pshapiro4broad — COMMENTED** [review]
The code looks OK, there are a few more UI changes that need to be added though.

File: src/pages/library/Datasets.js (hunk 2 above)
Comment (nawatts) [reply src/pages/library/Datasets.js]: >  We can't put useStore(authStore) in Datasets. I think we tried that first and it didn't work, which is why we moved to this approach. We wanted a complete re-instantiation of the component, rather than a partial re-rendering.

Both updating component state and updating state in the wrapper component would re-render the component. If you want to create a new component instance, you can change the component's `key`. 
https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key

File: src/pages/library/Datasets.js (hunk 4 above)
Comment (nawatts) [reply src/pages/library/Datasets.js:476]: Yeah, on second thought, keeping the links in the sidebar and on the front page and then prompting to login when a user clicks the link makes sense.

File: src/pages/library/Datasets.js (hunk 2 above)
Comment (s-rubenstein) [reply src/pages/library/Datasets.js]: My understanding was that state updates triggered a smart re-render, basically that react made a best effort guess on what should or should not re-render when state was updated, where as props triggered a re-render of the whole component. However, yeah, it wouldn't like… completely shut down and recreate the component from scratch.

Either way, this is fine as it is now.

Comment (nawatts) [reply src/pages/library/Datasets.js]: A change to either props or state causes the component to render. React then reconciles the DOM with the new element tree returned by that render.

https://reactjs.org/docs/faq-state.html#what-is-the-difference-between-state-and-props
https://reactjs.org/docs/reconciliation.html

File: src/libs/config.js [hunk 13]
```diff
@@ -11,8 +11,7 @@ export const getConfig = () => {
 export const isAnvil = () => (window.location.hostname === 'anvil.terra.bio') || getConfig().isAnvil
 export const isAnalysisTabVisible = () => getConfig().isAnalysisTabVisible
 export const isCromwellAppVisible = () => getConfig().isCromwellAppVisible
-// configOverridesStore.set({ isDataBrowserVisible: true }) in browser console to enable
-export const isDataBrowserVisible = () => getConfig().isDataBrowserVisible
+export const isDataBrowserFrontPage = () => false
```
Comment (pshapiro4broad) [inline src/libs/config.js:14]: If there isn't one already, can you make a ticket for the work to show the front page banner, or to remove it if we decide not to include it after all?

**Review by pshapiro4broad — APPROVED** [review]
LGTM!

File: src/pages/library/DataBrowserToggler.js [hunk 14]
```diff
@@ -97,30 +89,23 @@ export const DataBrowserPreviewToggler = ({ checked }) => {
...
       icon('talk-bubble', { size: 45, style: { marginLeft: '1.5rem', margin: '0 0.5rem' } }),
       div({ style: { display: 'flex', flexDirection: 'column' } }, [
-        strong('Provide feedback'),
+        'Provide feedback or looking for a dataset not listed?',
         h(Link, {
           onClick: () => setFeedbackShowing(true),
           style: { display: 'block', marginTop: 10 }
         },
-        ['What do you think about the new Data Catalog?'])
+        ['Let us know!'])
```
Comment (nawatts) [inline src/pages/library/DataBrowserToggler.js:43]: It would be nice to add an `aria-label` to this link since "Let us know!" isn't descriptive without the context of "Provide feedback or looking for a dataset not listed?".

File: src/pages/library/DataBrowserToggler.js [hunk 15]
```diff
@@ -142,6 +127,10 @@ export const DataBrowserPreviewToggler = ({ checked }) => {
     [div({ style: { fontWeight: 600, fontSize: 18 } },
       'Thank you for helping us improve the Data Catalog experience!')]
     )
-  ])
+  ]), !catalogShowing ? div({
+    style: {
+      padding: '15px 15px', wordWrap: 'break-word', width: 300, lineHeight: 1.5
```
Comment (nawatts) [inline src/pages/library/DataBrowserToggler.js:56]: Why limit the width of this and make the text wrap?

Screen Shot 2022-07-05 at 3 41 32 PM

**Review by nawatts — APPROVED** [review]

File: src/pages/library/DataBrowserToggler.js (hunk 15 above)
Comment (tcjiang9) [reply src/pages/library/DataBrowserToggler.js:56]: just to match our UI mock
https://projects.invisionapp.com/d/main?origin=v7#/console/22047689/467561712/preview

**Review by fboulnois — APPROVED** [review]
lgtm
